### PR TITLE
Simplified version of Block Clock

### PR DESCRIPTION
**Pull request by shaavan**
This PR is the simplified version of #148. And removes commits in #127 to allow easier upstream commit merges.

Functional changes from the original version:

1. The **PeersIndicator** has been replaced with plain rectangles to keep the UI intact.
2. The `conns` property representing if the number of connections is greater than 0, has been set to always positive. With this change, the "connecting" page, though implemented, becomes inaccessible through the UI.

### CODE REVIEW

**Comment by johnny9** [comment]
On Ubuntu 22.04, I had to install `apt install qml-module-qtgraphicaleffects` for the application to load otherwise I receive the error 
```
QQmlApplicationEngine failed to load component
qrc:/qml/pages/main.qml:53:13: Type NodeRunner unavailable
qrc:/qml/pages/node/NodeRunner.qml:24:9: Type BlockClockComponent unavailable
qrc:/qml/components/BlockClockComponent.qml:6:1: Type BlockClock unavailable
qrc:/qml/controls/BlockClock.qml:4:1: module "QtGraphicalEffects" is not installed
```

Might want to extend the README for the qml directory to include that module.

**Comment by johnny9** [comment]
It looks great.  Tested a sync on signet and here's what I got.

|Screenshot_2022-12-16_22-18-31|Screenshot_2022-12-16_22-19-02|Screenshot_2022-12-16_22-25-01|Screenshot_2022-12-16_22-26-42|Screenshot_2022-12-16_22-28-51|
|---|---|---|---|---|

File: src/qml/chainmodel.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+private:
+    QVariantList m_time_ratio_list{0.0};
+
+    QTimer* timer;
+
+    interfaces::Chain& m_chain;
+};
+
+#endif // BITCOIN_QML_CHAINMODEL_H
```
Comment (johnny9) [inline src/qml/chainmodel.h:50]: "No newline at end of file"

File: src/qml/pages/node/NodeRunner.qml [hunk 2]
```diff
@@ -18,26 +18,9 @@ Page {
...
-            ProgressIndicator {
-                width: 200
-                Layout.alignment: Qt.AlignCenter
-                progress: nodeModel.verificationProgress
-            }
-        }
+
+        Component.onCompleted: nodeModel.startNodeInitializionThread();
+
+        BlockClockComponent{}
```
Comment (johnny9) [inline src/qml/pages/node/NodeRunner.qml:24]:
```
GUI: qrc:/qml/pages/node/NodeRunner.qml:24:9: QML BlockClockComponent: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
```
Should be able to remove the ColumnLayout containing the BlockClockComponent and just define the BlockClockComponent as the content to this Page.

File: src/qml/controls/BlockClock.qml [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                ctx.arc(x, y, (width / 2) - parent.lineWidth / 2, start, end, false)
+                ctx.lineWidth = root.lineWidth
+                ctx.strokeStyle = root.colorCircle
+                ctx.stroke()
+            }
+
+            else {
+                var del = 0.0025
+                // Paint Block time points
+                for (var i=1; i<parent.blockList.length - 1; i++) {
```
Comment (johnny9) [inline src/qml/controls/BlockClock.qml:96]: The following error appears when going into the paused state.

```
2022-12-17T03:50:17Z GUI: qrc:/qml/controls/BlockClock.qml:96: TypeError: Value is undefined and could not be converted to an object
```

File: src/qml/components/BlockClockComponent.qml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+        State {
+            name: "blockClock"; when: synced && !pause && conns
+            PropertyChanges {
+                target: blockClock
+
+                ringProgress: blockList[0]
+                header: Math.floor(nodeModel.blockTipHeight / 1000) + "," + nodeModel.blockTipHeight % 1000
+                subText: "Blocktime"
+                blockList: chainModel.timeRatioList
```
Comment (johnny9) [inline src/qml/components/BlockClockComponent.qml:34]: QML error appears when shutting down.

```
2022-12-17T03:52:37Z Shutdown: In progress...
2022-12-17T03:52:37Z torcontrol thread exit
2022-12-17T03:52:37Z addcon thread exit
2022-12-17T03:52:37Z msghand thread exit
2022-12-17T03:52:37Z net thread exit
2022-12-17T03:52:39Z opencon thread exit
2022-12-17T03:52:39Z DumpAnchors: Flush 0 outbound block-relay-only peer addresses to anchors.dat started
2022-12-17T03:52:39Z DumpAnchors: Flush 0 outbound block-relay-only peer addresses to anchors.dat completed (0.01s)
2022-12-17T03:52:39Z scheduler thread exit
2022-12-17T03:52:39Z Writing 0 unbroadcast transactions to disk.
2022-12-17T03:52:39Z Dumped mempool: 3.7e-06s to copy, 0.00448635s to dump
2022-12-17T03:52:39Z Shutdown: done
2022-12-17T03:52:39Z GUI: qrc:/qml/components/BlockClockComponent.qml:34: TypeError: Cannot read property 'timeRatioList' of null
```

File: src/qml/chainmodel.cpp [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+void ChainModel::setCurrentTimeRatio()
+{
+    int secsSinceMeridian = (QTime::currentTime().msecsSinceStartOfDay() / 1000) % SECS_IN_12_HOURS;
+    double currentTimeRatio = double(secsSinceMeridian) / SECS_IN_12_HOURS;
+
+    if (currentTimeRatio < m_time_ratio_list[0].toDouble()) { // That means time has crossed a meridian
+        m_time_ratio_list.erase(m_time_ratio_list.begin() + 1, m_time_ratio_list.end());
+    }
+    m_time_ratio_list[0] = currentTimeRatio;
+}
```
Comment (johnny9) [inline src/qml/chainmodel.cpp:75]: "No new line at end of file"

**Review by johnny9 — CHANGES_REQUESTED** [review]
Some QML errors appear during runtime. `tail -f debug.log | grep GUI` to watch for issues when running bitcoin-qt.

File: src/qml/controls/BlockClock.qml (hunk 3 above)
Comment (johnny9) [reply src/qml/controls/BlockClock.qml:96]: Also a small style thing, most prefer to have spaces before and after operators so its easier to read.

**Comment by shaavan** [comment]
Moved the changes of this PR to #148 to preserve comment history for the block_clock and prevent the splitting of discussion between two PRs.

@johnny9, I shall be addressing your comments here and in #148. Thanks for your cooperation.

**Pull request closed without merging**